### PR TITLE
Cleanup velox/parse/Expressions.h

### PR DESCRIPTION
**Pull request by mbasmanova**
Remove not used not tested APIs: serialize + create, withInputs,
equalsNonRecursive methods and SortExpr class.

### CODE REVIEW

File: velox/parse/Expressions.h [hunk 1]
```diff
@@ -87,33 +82,7 @@ class InputExpr : public core::IExpr {
     return EMPTY();
   }
 
-  std::shared_ptr<const IExpr> withInputs(
```
Comment (pedroerp) [inline velox/parse/Expressions.h:90]: Was this just legacy from Koski?

File: velox/parse/Expressions.h [hunk 2]
```diff
@@ -205,63 +141,6 @@ class FieldAccessExpr : public core::IExpr {
   VELOX_DEFINE_CLASS_NAME(FieldAccessExpr)
 };
 
-class SortExpr : public core::IExpr {
```
Comment (pedroerp) [inline velox/parse/Expressions.h:208]: out of curiosity: what's the "correct" way to represent sorting from a parsed SQL statement?

**Review by pedroerp — APPROVED** [review]
A few questions for my own understanding, but thanks for the cleanup!

File: velox/parse/Expressions.h (hunk 1 above)
Comment (mbasmanova) [reply velox/parse/Expressions.h:90]: That's right. This is a copy-paste from Koski.

File: velox/parse/Expressions.h (hunk 2 above)
Comment (mbasmanova) [reply velox/parse/Expressions.h:208]: There is a special API for that: parseOrderByExpr

```
// Parses an ORDER BY clause using DuckDB's internal postgresql-based parser,
// converting it to a pair of an IExpr tree and a core::SortOrder. Uses ASC
// NULLS LAST as the default sort order.
std::pair<std::shared_ptr<const core::IExpr>, core::SortOrder> parseOrderByExpr(
    const std::string& exprString);
```

**Pull request closed without merging**